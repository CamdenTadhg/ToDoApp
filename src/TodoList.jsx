import react, {useState} from 'react';
import NewTodoForm from './NewTodoForm.jsx';
import Todo from './Todo.jsx';
import {v4 as uuid} from 'uuid';
import React from 'react';

const TodoList = () => {
    const INITIAL_STATE = [
        {id: uuid(), task: 'Take pills'},
        {id: uuid(), task: 'Prep for A'},
        {id: uuid(), task: 'Breakfast with yogurt'}
    ];
    const {todos, setTodos} = useState(INITIAL_STATE);

    const addTodo = (task) => {
        let todosCopy = todos.map(t => ({...t}));
        todosCopy.push({id: uuid(), task});
        setTodos(todosCopy);
    }

    const deleteTodo = (id) => {
        let todosCopy = todos.map(t => ({...t}));
        todosCopy.filter(t => t.id !== id);
        setTodos(todosCopy);
    }

    return (
        <React.Fragment>
            <NewTodoForm addTodo={addTodo}/>
            {todos.map(t => {
                <Todo key={t.id} id={t.id} task={t.task} deleteTodo={deleteTodo}/>
            })}
        </React.Fragment>
    )
}

export default TodoList;