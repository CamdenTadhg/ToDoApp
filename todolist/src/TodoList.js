import react, {useState} from 'react';
import NewTodoForm from './NewTodoForm.js';
import Todo from './Todo.js';
import {v4 as uuid} from 'uuid';
import React from 'react';

const TodoList = () => {
    const INITIAL_STATE = [
        {id: uuid(), task: 'Take pills'},
        {id: uuid(), task: 'Prep for A'},
        {id: uuid(), task: 'Breakfast with yogurt'}
    ];
    const [todos, setTodos] = useState(INITIAL_STATE);

    const addTodo = (task) => {
        let todosCopy = todos.map(t => ({...t}));
        todosCopy.push({id: uuid(), task});
        setTodos(todosCopy);
    }

    const deleteTodo = (id) => {
        const todosCopy = todos.filter(t => t.id !== id);
        setTodos(todosCopy);
    }

    return (
        <React.Fragment>
            <NewTodoForm addTodo={addTodo}/>
            {todos.map(t => (
                <Todo key={t.id} id={t.id} task={t.task} deleteTodo={deleteTodo}/>
            ))}
        </React.Fragment>
    )
}

export default TodoList;