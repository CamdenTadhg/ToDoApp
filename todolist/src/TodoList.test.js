import react from 'react';
import {render} from '@testing-library/react';
import TodoList from './TodoList.js';

it('should render correctly', function() {
    render(<TodoList/>);
});

it('should match snapshot', function() {
    const {asFragment} = render(<TodoList/>);
    expect(asFragment()).toMatchSnapshot();
});