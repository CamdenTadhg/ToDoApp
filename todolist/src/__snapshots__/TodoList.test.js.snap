// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form
    data-testid="todoform"
  >
    <label
      for="todo"
    >
      Task:
      <input
        id="todo"
        name="todo"
        placeholder="task"
        type="text"
        value=""
      />
    </label>
    <button>
      Create ToDo
    </button>
  </form>
  <div>
    Take pills
    <button
      data-testid="Take pills"
    >
      X
    </button>
  </div>
  <div>
    Prep for A
    <button
      data-testid="Prep for A"
    >
      X
    </button>
  </div>
  <div>
    Breakfast with yogurt
    <button
      data-testid="Breakfast with yogurt"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
